crates/re_ui/examples/re_ui_example.rs

```diff
@@ -600,3 +647,751 @@ impl egui_tiles::Behavior<Tab> for MyTileTreeBehavior {
+            } else {
+                None
+            };
+
+            if let Some(previous_container_id) = previous_container_id {
+                Some(DragTarget::new(
+                    top.top(),
+                    (response.rect.left() + indent..=response.rect.right()).into(),
+                    previous_container_id,
+                    usize::MAX,
+                ))
+            } else {
+                Some(DragTarget::new(
+                    top.top(),
+                    response.rect.x_range(),
+                    parent_id,
+                    pos_in_parent,
+                ))
+            }
+        } else if !is_container {
+            if ui.rect_contains_pointer(bottom) {
+                // insert after me
+                Some(DragTarget::new(
+                    bottom.bottom(),
+                    response.rect.x_range(),
+                    parent_id,
+                    pos_in_parent + 1,
+                ))
+            } else {
+                None
+            }
+        } else {
+            let body_rect = body_response.map(|r| r.rect).filter(|r| r.width() > 0.0);
+            if let Some(body_rect) = body_rect {
+                if ui.rect_contains_pointer(left_bottom) || ui.rect_contains_pointer(bottom) {
+                    // insert at pos = 0 inside me
+                    Some(DragTarget::new(
+                        left_bottom.bottom(),
+                        (body_rect.left() + indent..=body_rect.right()).into(),
+                        item_id,
+                        0,
+                    ))
+                } else if ui.rect_contains_pointer(content_left_bottom) {
+                    // insert after me in my parent
+                    Some(DragTarget::new(
+                        content_left_bottom.bottom(),
+                        response.rect.x_range(),
+                        parent_id,
+                        pos_in_parent + 1,
+                    ))
+                } else {
+                    None
+                }
+            } else if ui.rect_contains_pointer(left_bottom) {
+                // insert after me in my parent
+                Some(DragTarget::new(
+                    left_bottom.bottom(),
+                    response.rect.x_range(),
+                    parent_id,
+                    pos_in_parent + 1,
+                ))
+            } else if ui.rect_contains_pointer(bottom) {
+                // insert at pos = 0 inside me
+                Some(DragTarget::new(
+                    bottom.bottom(),
+                    (response.rect.left() + indent..=response.rect.right()).into(),
+                    item_id,
+                    0,
+                ))
+            } else {
+                None
+            }
+        }
+    }
+}
+
+struct DragTarget {
+    /// Y coordinate for the cursor
+    insert_y: f32,
+    /// Range of X coordinates for the cursor
+    range_x: egui::Rangef,
```

Comment:
So this effectively forms a horizontal line - why not use a Rect here?

Reply:
Because a rect is one more parameter than I need, and is thus ambiguous?